File: java/behavior/strategy/src/entities/MallardDuck.java
package java.behavior.strategy.src.entities;
public class MallardDuck extends Duck{
    public MallardDuck(FlyWithWings fw, Quack quack){
        super(fw, quack);
    }

    public void display(){
        System.out.println("Hi, im a Mallard Duck");
    }
}

File: c++/behavior/strategy/src/entities/MallardDuck.cpp
#include <iostream>
#include "../headers/MallardDuck.hpp"
#include "../headers/FlyWithWings.hpp"
#include "../headers/Quack.hpp"
using namespace std;

MallardDuck::MallardDuck() : Duck(FlyWithWings(), Quack()){
	cout << "Mallard Pato contruido" << endl;
}

void MallardDuck::display(){
	cout << "Pato mallard aparecendo" << endl;
}